core/shared/src/main/scala/org/http4s/Header.scala

```diff
@@ -170,20 +170,28 @@ object Header {
         h: H
     )(implicit H: Header[H, _]): Header.ToRaw with Primitive =
       new Header.ToRaw with Primitive {
-        val values = List(Header.Raw(H.name, H.value(h)))
+        val values = Header.Raw(H.name, H.value(h)) :: Nil
       }
 
     implicit def foldablesToRaw[F[_]: Foldable, H](
         h: F[H]
     )(implicit convert: H => ToRaw with Primitive): Header.ToRaw = new Header.ToRaw {
-      val values: List[Raw] = h.toList.foldMap(v => convert(v).values)
+      val values = h
+        .foldLeft(collection.mutable.ListBuffer.empty[Header.Raw]) { (buf, v) =>
+          buf ++= convert(v).values
+        }
+        .toList
     }
 
     // Required for 2.12 to convert variadic args.
     implicit def scalaCollectionSeqToRaw[H](
         h: collection.Seq[H]
     )(implicit convert: H => ToRaw with Primitive): Header.ToRaw = new Header.ToRaw {
-      val values: List[Raw] = h.toList.foldMap(v => convert(v).values)
+      val values = h
+        .foldLeft(collection.mutable.ListBuffer.empty[Header.Raw]) { (buf, v) =>
+          buf ++= convert(v).values
+        }
+        .toList
```

Comment:
Can we use `foreach` for this one? I expect that would be faster than `foldLeft` but I don't know for sure.